### PR TITLE
TC per-lcore tuning

### PR DESCRIPTION
**Pull request by ywc689**
- Refactor TC module to be per-lcore.
- Fix crash problem when add/modify/del qsch/qcls dynamically with traffic bursting in.
- Fix inaccuracy traffic control problem of tbf caused by multi-thread contention of tokens.
- Schedule qsch per loop to make TC work properly.

### CODE REVIEW

**Comment by haosdent** [comment]
Would this make QoS usable?

**Comment by ywc689** [comment]
> Would this make QoS usable?

Just some simple cases, such as traffic shaping and policing. I'll write a doc to show some use cases later.

**Comment by haosdent** [comment]
I think, thanks a lot! @ywc689

File: include/linux_ipv6.h [hunk 1]
```diff
@@ -266,6 +266,18 @@ static inline int ipv6_addr_cmp(const struct in6_addr *a1, const struct in6_addr
     return memcmp(a1, a2, sizeof(struct in6_addr));
 }
 
+static inline int ipv6_addr_cmp_u128(const struct in6_addr *a1, const struct in6_addr *a2)
```
Comment (yifan1227) [inline include/linux_ipv6.h:269]: What's the difference between thiss function and the upper one?

Comment (yifan1227) [inline include/linux_ipv6.h:269]: This function is duplicated with the upper one.

**Review by yifan1227 — APPROVED** [review]

**Pull request merged**